src/elements/Tree.tsx

```diff
@@ -0,0 +1,269 @@
+import Link from "@vertigis/web/ui/Link";
+import List, { ListProps } from "@vertigis/web/ui/List";
+import ListItemIcon from "@vertigis/web/ui/ListItemIcon";
+import ListItemText from "@vertigis/web/ui/ListItemText";
+import ListItem from "@vertigis/web/ui/ListItem";
+
+/**
+ * Properties for the `Tree` component.
+ */
+export declare type TreeViewProps = {
+    showCheckBoxes?: boolean;
+};
+
+const IconButtonStyleOverrides: IconButtonProps["sx"] = {
+    // Hack required to override .gcx-forms.defaults
+    borderRadius: "50% !important",
+    // Hack required to match disabled list item content
+    "&.Mui-disabled": {
+        opacity: 0.38,
+    },
+};
+
+type SettableBoxProps = Pick<BoxProps, "maxHeight" | "maxWidth">;
+
+type SettableTreeProps = Pick<ListProps, "dense" | "subheader">;
+
+type SettableCheckBoxProps = Pick<TreeViewProps, "showCheckBoxes">;
+
+interface TreeElementNode {
+    key: string;
+    open: boolean;
+    primary: string;
+    secondary?: string;
+    link?: string;
+    icon?: DynamicIconProps["src"];
+    checked?: boolean;
+    children?: TreeElementNode[];
+}
+interface TreeElementProps
+    extends FormElementProps<TreeElementNode[] | undefined>,
+        SettableBoxProps,
+        SettableTreeProps,
+        SettableCheckBoxProps {
+    items: TreeElementNode[];
+    onClick?: (node: TreeElementNode) => void;
+    onMouseEnter?: (node: TreeElementNode) => void;
+    onMouseLeave?: (node: TreeElementNode) => void;
+}
+
+function flattenNodes(node: TreeElementNode, collection: TreeElementNode[]) {
+    if (node.checked) {
+        collection.push(node);
+    }
+    if (node.children && node.children.length > 0) {
+        for (const child of node.children) {
+            flattenNodes(child, collection);
+        }
+    } else {
+        return;
+    }
+}
+
+function filterTree(nodes: TreeElementNode[]) {
+    const children: TreeElementNode[] = [];
+    for (const node of nodes) {
+        flattenNodes(node, children);
+    }
+    return children;
+}
+
+function renderChild(
+    node: TreeElementNode,
+    index: number,
+    value: TreeElementNode[] | undefined,
+    showCheckBoxes: boolean | undefined,
+    level: number,
+    handleMouseEnter: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleMouseLeave: (event: React.MouseEvent, node: TreeElementNode) => void,
+    handleItemClick: (node: any) => void
+) {
+    const isFolder = Array.isArray(node.children);
+    const isNode = !node.children;
+    const m = 1.5 * level;
+    const icon = (
+        <ChevronDownIcon
+            fontSize="small"
+            sx={{
+                transform: `rotate(${node.open ? 0 : -90}deg)`,
+                transition: "transform 0.2s ease-in-out",
+            }}
+        />
+    );
+    return (
+        <React.Fragment>
+            <ListItem key={node.key} sx={{ paddingLeft: (theme) => theme.spacing(m) }}>
+                {isNode && showCheckBoxes && (
```

Comment:
We are missing some hover/click state that Web has in its layer list. Web everything in a `<ListItemButton>` to get that.

Reply:
I think I've got it pretty close to Web now.